lib/miou_unix.mli

```diff
@@ -20,7 +20,7 @@ val tcpv4 : unit -> file_descr
 val tcpv6 : unit -> file_descr
 (** [tcpv6 ()] allocates a new IPv6 socket. *)
 
-val bind_and_listen : ?backlog:int -> file_descr -> Unix.sockaddr -> unit
+val bind_and_listen : ?backlog:int -> ?reuseaddr:bool -> ?reuseport:bool -> file_descr -> Unix.sockaddr -> unit
```

Comment:
I wonder, is there direct access to the file descriptor passed into this function? Then a user could just set their own socket options... TCP_NODELAY is likely another thing some people may want to set ;)

Reply:
There's direct access through `Miou_unix.to_file_descr`. We could always just remove the setsockopts from bind_and_listen altogether; libraries like httpcats already provide a wrapper around this, so it's a low level enough interface to leave it up to the user.

Reply:
also: I would suggest in any case providing a Miou_unix.setsockopt function instead of making the user grab the real fd